### PR TITLE
Adding sections to Readme.md

### PR DESCRIPTION
**Pull request by jossiwolf**
Significant changes: 
* Added a ["First Steps" section](https://github.com/jossiwolf/Android-Study-Jam#first-steps) in the Readme.md which I will update over time
* Added a ["Good-to-have-but-not-needed previous knowledge" section](https://github.com/jossiwolf/Android-Study-Jam/blob/master/README.md#good-to-have-but-not-needed-previous-knowledge-especially-this-is-open-for-discussion---) - the content of this has to be discussed!
* Added a ["Questions to either talk about or address in the guides" section](https://github.com/jossiwolf/Android-Study-Jam#questions-to-either-talk-about-or-address-in-the-guides) which contains questions that may come up
* Updated the build tools version, target SDK as well as support library version to 25 and 25.1.0
* Added my name to the license file

Minor changes:
* Changed the order of no 3 & 4 under ["Proposed feature steps"](https://github.com/jossiwolf/Android-Study-Jam#proposed-feature-steps) as creating chat groups is most likely harder than adding image support (which is very easy to add)
* Added some keywords in the ["Proposed feature steps" section](https://github.com/jossiwolf/Android-Study-Jam#proposed-feature-steps)

Concerning **_iv._** and _**v.**_ under '_Firebase Setup'_ (https://github.com/jossiwolf/Android-Study-Jam/#first-steps), we'll have to decide between the Android Studio Firebase Plugin and setting up everything manually.  I think you can get a better understanding of how things work when you do it yourself, e.g. that `google-services.json` contains the relevant data for Firebase / Google Services.

### CODE REVIEW

**Comment by jossiwolf** [comment]
There's something missing, sorry!

**Comment by jossiwolf** [comment]
Now it's alright!

File: README.md [hunk 1]
```diff
@@ -1,20 +1,53 @@
...
 This is the reference app that we want to build during the course. The idea is a simple WhatsApp like chat application that uses Firebase as its backend.
 
 ### Proposed feature steps
 Not all must be accomplished
 
-1. Send end receive text messages and display them in a list.
+1. Send and receive text messages and display them in a list (Recycler View).
 
-2. Make users identifiable by a name
+2. Make users identifiable by a name (Firebase Authentication)
```
Comment (flbaue) [inline README.md]: I thought to first of all just let the user set a name somewhere and use it for the messages. The Firebase Authentication part was rather meant to be the login step.
I guess people should just play around with the Firebase DB first and later start to use different Firebase Services

**Comment by flbaue** [comment]
Besides my comment above 👍

**Comment by jossiwolf** [comment]
@flbaue Yeah, I get your point. But if there is no authentication by id, it is possible that there are mix-ups.
I think it doesn't really matter, because it's either: 
* Trying to insert into the DB and then checking if a node with the user's name exists
* Using Firebase Auth
With these two options, I think that Firebase Authentication would be easier to use. But maybe you had something else in mind when thinking about the DB structure?

**Comment by flbaue** [comment]
I did not even have something that advanced in mind. A name will never be unique and I would not think that far. its just that there are names with the messages somehow. 
I think that Firebase Auth is actually not that simple for a beginner, since it introduces a lot of complexity (authentication, logged in or not, auth providers, program flow, etc). Therefor I actually totally agree to put login as the last step, as you did :-)

For example sending images, which means uploading it to firebase storage, getting the id, sending the id, and than retrieving the image again on the receiver side is much simpler since it is closer to how normal messages would be handled

**Comment by jossiwolf** [comment]
Yep, perfect!

**Pull request merged**